src/main/java/org/takes/misc/Utf8InputStreamReader.java

```diff
@@ -0,0 +1,51 @@
+/**
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2014-2016 Yegor Bugayenko
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.takes.misc;
+
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.charset.Charset;
+
+/**
+ * InputStreamReader that uses UTF-8 encoding for all operations.
+ * @author Dali Freire (dalifreire@gmail.com)
+ * @version $Id$
+ * @since 0.33
+ */
+public class Utf8InputStreamReader extends InputStreamReader {
+
+    /**
+     * UTF-8 encoding.
+     */
+    private static final String ENCODING = "UTF-8";
+
+    /**
+     * Ctor.
+     * @param input InpputStream value
+     */
+    public Utf8InputStreamReader(final InputStream input) {
+        super(input, Charset.forName(Utf8InputStreamReader.ENCODING));
```

Comment:
@dmzaytsev I thought that if we use StandardCharsets.UTF_8 we will increasing the coupling. But I agree with you that this can be a good approach.